src/pages/page/[pageId].tsx

```diff
@@ -29,29 +37,64 @@ interface Props {
   chaptersData: ChaptersData;
 }
 
-const QuranicPage: NextPage<Props> = ({ hasError, pageVerses }) => {
+const QuranicPage: NextPage<Props> = ({ hasError, pageVerses: initialData }) => {
   const { t, lang } = useTranslation('common');
   const {
     query: { pageId },
   } = useRouter();
 
   const mushafId = useGetMushaf();
+  const {
+    data: pageVersesData,
+    isLoading: isPageVersesLoading,
+    error: pageVersesError,
+  } = useFetchPageVerses(String(pageId), initialData);
+
+  const isUsingDefaultFont = useSelector(selectIsUsingDefaultFont);
+  const quranReaderStyles = useSelector(selectQuranReaderStyles, shallowEqual);
 
-  if (hasError || pageId > PAGES_MUSHAF_MAP[Number(mushafId)]) {
+  const {
+    lookupRange: pagesLookupData,
+    isLoading: isPagesLookupLoading,
+    hasError: pagesLookupError,
+  } = useFetchPagesLookup(
+    String(pageId),
+    QuranReaderDataType.Page,
+    initialData,
+    quranReaderStyles,
+    isUsingDefaultFont,
+  );
+
+  if (
+    hasError ||
+    pageId > PAGES_MUSHAF_MAP[Number(mushafId)] ||
+    pagesLookupError ||
+    pageVersesError
+  ) {
     return <Error statusCode={500} />;
   }
 
+  if (isPageVersesLoading || isPagesLookupLoading) {
+    return <Spinner />;
+  }
+
   const path = getPageNavigationUrl(Number(pageId));
+  const data = {
+    ...pageVersesData,
+    pageVerses: { pagesLookup: pagesLookupData },
+    metaData: { numberOfVerses: pageVersesData.verses.length },
+  };
```

Comment:
Since this is common with `getStaticProps`, we should create a common util function between both.